sufia-models/lib/sufia/models/utils.rb

```diff
@@ -1,12 +1,26 @@
+# This module includes a utility methods
 module Sufia
   module Utils
     extend ActiveSupport::Concern
 
+    # retry the block if the conditional call is true unless we hit the maximum tries
+    #
+    # @param number_of_tries [Fixnum] maximum number of times to retry the block
+    # @param condition [Boolean] conditional to call and see if we SHOULD retry
```

Comment:
For the condition param, we expect not to have a Boolean passed but rather something callable. Maybe this should be `s/Boolean/#call/`